parse_snyk_output.cjs

```diff
@@ -0,0 +1,178 @@
+                    },
+                  };
+
+                  
+
+                  let issueResponse;
+
+                  try {
+                    const jiraUrl = `${process.env.JIRA_BASE_URL}/rest/api/2/issue`;
+                    console.log('JIRA_URL:', jiraUrl);
+                    issueResponse = await jira.addNewIssue(issue);
+                    console.log(issueResponse)
+                    console.log(`Jira ticket created for vulnerability: ${vulnerability.title}`);
+                  } catch (error) {
+                    console.error('Error creating Jira ticket:', error);
+                  }
+
+                  // attach the scan report to the Jira ticket
+                  const attachmentUrl = `https://${process.env.JIRA_BASE_URL}/rest/api/2/issue/${issueResponse.key}/attachments`;
+                  const attachment = {
+                    method: 'POST',
+                    uri: attachmentUrl,
+                    auth: {
+                      username: process.env.JIRA_USER_EMAIL,
+                      password: process.env.JIRA_API_TOKEN
+                    },
+
+                    headers: {
+                      'X-Atlassian-Token': 'no-check', // Disable XSRF check for file upload
+                      'X-Requested-With': 'XMLHttpRequest',
+                      'X-Scheme': 'https',
+                      'X-Forwarded-Proto': 'https',
+                    },
+
+                    formData: {
+                      file: fs.createReadStream('snyk_output.txt')
+                    },
+                  };
+                  
+                  try {
+                    await rp(attachment);
+                  } catch (error) {
+                    console.error(`Error attaching file to Jira ticket: ${error}`);
+                    return;
+                  }
+                  
+                  console.log(`Jira ticket ${issueResponse.key} created successfully.`);
+                  return issueResponse;
+      }
+}
+
+
+          (async () => {
+            const consoleOutputFile = process.argv[2];
+            const jsonData = fs.readFileSync(consoleOutputFile, 'utf-8');
+
+            const vulnerabilities = parseSnykOutput(jsonData);
+            // console.log("Vulnerabilities:", JSON.stringify(vulnerabilities, null, 2));
+            console.log(`Parsed vulnerabilities: ${vulnerabilities.length}`);
+
+            const uniqueVulnerabilities = Array.from(new Set(vulnerabilities.map(v => v.title)))
+              .map(title => {
+                return vulnerabilities.find(v => v.title === title);
+              });
+
+            const seenTitles = new Set();
+
+            for (const vulnerability of uniqueVulnerabilities) {
+              if (seenTitles.has(vulnerability.title)) {
+                console.log(`Skipping duplicate vulnerability: ${vulnerability.title}`);
+                continue;
+              }
+              seenTitles.add(vulnerability.title);
```

Comment:
Request: This logic is redundant. If we've already filtered the vulnerabilities to be unique by title, then there is no need to keep track of the `seenTitles`. Let's do one or the other.

Reply:
Done!